### PR TITLE
Add prebuilt intel tf build option

### PR DESCRIPTION
**Pull request by kanvi-nervana**
depends on #684

### CODE REVIEW

**Comment by adk9** [comment]
Can we just have `--use-prebuilt-tf` take a package/version defaulting to `tensorflow==2.2.0`?

**Comment by kanvi-nervana** [comment]
> Can we just have `--use-prebuilt-tf` take a package/version defaulting to `tensorflow==2.2.0`?

We already have the option to take a version, in addition to that an option to differentiate between native or intel tf? Isn't having a separate build flag more straight forward?

**Comment by adk9** [comment]
> > Can we just have `--use-prebuilt-tf` take a package/version defaulting to `tensorflow==2.2.0`?
> 
> We already have the option to take a version, in addition to that an option to differentiate between native and intel tf? Isn't having a separate build flag more straight forward?

The `--use-prebuilt-tf` flag simply instructs the script to build against a prebuilt TF wheel that comes from PyPI. It doesn't strictly matter where it comes from or what version it is. (It may matter now if you need to fetch the source to run TF unit tests but it's possible to extract the version of the prebuilt package).

This is okay but I was thinking of something more general that allows us to choose packages or optionally specify version numbers: 
* `--use-prebuilt-tf=tensorflow`
* `--use-prebuilt-tf=intel-tensorflow`
* `--use-prebuilt-tf=intel-tensorflow==2.3.0`

so on and so forth.

**Comment by bani-intelaipg** [comment]
I have similar comment like adk9

File: build_ngtf.py [hunk 1]
```diff
@@ -29,7 +29,8 @@ def version_check(use_prebuilt_tensorflow, use_tensorflow_from_location,
             gcc_ver = get_gcc_version()
             if gcc_ver < '5.3.0':
                 raise Exception(
-                    "Need GCC 5.3.0 or newer to build using prebuilt TensorFlow\n"
+                    "Need GCC 5.3.0 or newer to build using prebuilt TensorFlow"
+                    " or Intel-Tensorflow\n"
```
Comment (adk9) [inline build_ngtf.py:33]:
```suggestion
                    " or Intel TensorFlow\n"
```

File: build_ngtf.py [hunk 2]
```diff
@@ -49,11 +50,10 @@ def version_check(use_prebuilt_tensorflow, use_tensorflow_from_location,
 
 def main():
     '''
-    Builds TensorFlow, ngraph, and ngraph-tf for python 3
+    Builds TensorFlow, OpenVino, and ngraph-tf for python 3
```
Comment (adk9) [inline build_ngtf.py:53]:
```suggestion
    Builds TensorFlow, OpenVINO, and ngraph-tf for python 3
```

File: build_ngtf.py [hunk 3]
```diff
@@ -216,6 +218,9 @@ def main():
     if arguments.use_prebuilt_tensorflow != '':
         tf_version = arguments.use_prebuilt_tensorflow
 
+    if arguments.use_intel_tensorflow != '':
+        use_intel_tf = True
```
Comment (adk9) [inline build_ngtf.py:222]: No need for this variable, use `arguments.use_intel_tensorflow` directly.

File: build_ngtf.py [hunk 4]
```diff
@@ -144,8 +146,8 @@ def main():
 
     assert not (
         arguments.use_tensorflow_from_location != '' and
-        arguments.use_prebuilt_tensorflow != ''
-    ), "\"use_tensorflow_from_location\" and \"use_prebuilt_tensorflow\" "
+        (arguments.use_prebuilt_tensorflow != '')
```
Comment (adk9) [inline build_ngtf.py:149]: Is this change needed?

File: build_ngtf.py [hunk 5]
```diff
@@ -256,10 +261,15 @@ def main():
         os.chdir(cwd)
     else:
         if arguments.use_prebuilt_tensorflow != '':
-            print("Using existing TensorFlow version", tf_version)
-            # Install TensorFlow
-            command_executor(
-                ["pip", "install", "-U", "tensorflow==" + tf_version])
+            print("Using TensorFlow version", tf_version)
+            if use_intel_tf:
+                print("Install Intel-Tensorflow")
```
Comment (adk9) [inline build_ngtf.py:266]:
```suggestion
                print("Install Intel TensorFlow")
```

File: build_ngtf.py [hunk 6]
```diff
@@ -256,10 +261,15 @@ def main():
...
+                command_executor(
+                    ["pip", "install", "-U", "intel-tensorflow==" + tf_version])
+            else:
+                print("Install native Tensorflow")
```
Comment (adk9) [inline build_ngtf.py:270]:
```suggestion
                print("Install native TensorFlow")
```

File: build_ngtf.py [hunk 7]
```diff
@@ -272,10 +282,6 @@ def main():
                           "https://github.com/tensorflow/tensorflow.git",
                           tf_version)
             os.chdir(pwd_now)
-            if not arguments.disable_cpp_api:
-                # Now build the libtensorflow_cc.so - the C++ library
-                build_tensorflow_cc(tf_version, tf_src_dir, artifacts_location,
-                                    target_arch, verbosity)
         else:
```
Comment (adk9) [inline build_ngtf.py:285]: Is this still needed to build CC API? How would someone build the CC API with prebuilt TF now?

**Review by adk9 — CHANGES_REQUESTED** [review]

File: build_ngtf.py (hunk 3 above)
Comment (kanvi-nervana) [reply build_ngtf.py:222]: This variable is needed for  build_tensorflow and build_tensorflow_cc functions to build with config = mkl option.

File: build_ngtf.py (hunk 7 above)
Comment (kanvi-nervana) [reply build_ngtf.py:285]: As discussed, I removed the dependence for prebuilt option on this flag to reduce the number of build flags we need to pass.

File: build_ngtf.py (hunk 4 above)
Comment (kanvi-nervana) [reply build_ngtf.py:149]: I think it's accidental change and not needed.

**Review by adk9 — APPROVED** [review]
